File: ADIP/sdks/java/AdipClient.java
package com.adip.sdk;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.time.Duration;
import java.util.Map;
import java.util.stream.Collectors;

/**
 * ADIP Java SDK (reference client).
 *
 * Uses the JDK 11+ HttpClient (no external dependencies). Returns raw JSON
 * strings; plug in Jackson/Gson to deserialize into typed models.
 *
 * <pre>{@code
 *   AdipClient client = new AdipClient("http://localhost:8000");
 *   String health = client.get("/health", Map.of());
 *   String artifact = client.get("/artifact-generation/projects/1/generate",
 *                                Map.of("artifact_type", "BRD"));
 * }</pre>
 */
public class AdipClient {

    private final String baseUrl;
    private final String apiPrefix;
    private final String token;
    private final HttpClient http;

    public AdipClient(String baseUrl) {
        this(baseUrl, "/api/v1", null);
    }

    public AdipClient(String baseUrl, String apiPrefix, String token) {
        this.baseUrl = baseUrl.replaceAll("/$", "");
        this.apiPrefix = apiPrefix;
        this.token = token;
        this.http = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(30))
                .build();
    }

    public String get(String path, Map<String, String> params) {
        return send("GET", path, params, null);
    }

    public String post(String path, String jsonBody) {
        return send("POST", path, Map.of(), jsonBody);
    }

    private String send(String method, String path, Map<String, String> params, String jsonBody) {
        String url = baseUrl + apiPrefix + path;
        if (params != null && !params.isEmpty()) {
            String qs = params.entrySet().stream()
                    .map(e -> enc(e.getKey()) + "=" + enc(e.getValue()))
                    .collect(Collectors.joining("&"));
            url += "?" + qs;
        }
        HttpRequest.Builder builder = HttpRequest.newBuilder(URI.create(url))
                .timeout(Duration.ofSeconds(30))
                .header("Content-Type", "application/json")
                .header("Accept", "application/json");
        if (token != null) {
            builder.header("Authorization", "Bearer " + token);
        }
        if ("POST".equals(method)) {
            builder.POST(HttpRequest.BodyPublishers.ofString(jsonBody == null ? "{}" : jsonBody));
        } else {
            builder.GET();
        }
        try {
            HttpResponse<String> resp = http.send(builder.build(),
                    HttpResponse.BodyHandlers.ofString());
            if (resp.statusCode() >= 400) {
                throw new RuntimeException("ADIP HTTP " + resp.statusCode() + " for "
                        + method + " " + path + ": " + resp.body());
            }
            return resp.body();
        } catch (Exception e) {
            throw new RuntimeException("ADIP request failed: " + e.getMessage(), e);
        }
    }

    private static String enc(String s) {
        return URLEncoder.encode(s, StandardCharsets.UTF_8);
    }

    // --- convenience helpers ---
    public String health() {
        return get("/health", Map.of());
    }

    public String listProjects() {
        return get("/projects", Map.of("size", "50"));
    }

    public String generateArtifact(int projectId, String artifactType) {
        return get("/artifact-generation/projects/" + projectId + "/generate",
                Map.of("artifact_type", artifactType));
    }

    public String orchestrate(String prompt) {
        return post("/orchestrator/run", "{\"prompt\":\"" + prompt.replace("\"", "\\\"") + "\"}");
    }
}

File: ADIP/sdks/csharp/AdipClient.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace Adip.Sdk
{
    /// <summary>
    /// ADIP C# SDK (reference client).
    ///
    /// Uses <see cref="HttpClient"/> (no external dependencies). Returns raw JSON
    /// strings; use System.Text.Json to deserialize into typed models.
    ///
    /// <code>
    ///   using var client = new AdipClient("http://localhost:8000");
    ///   var health = await client.GetAsync("/health");
    ///   var artifact = await client.GetAsync("/artifact-generation/projects/1/generate",
    ///                      new() { ["artifact_type"] = "BRD" });
    /// </code>
    /// </summary>
    public sealed class AdipClient : IDisposable
    {
        private readonly string _baseUrl;
        private readonly string _apiPrefix;
        private readonly HttpClient _http;

        public AdipClient(string baseUrl = "http://localhost:8000",
                          string apiPrefix = "/api/v1", string? token = null)
        {
            _baseUrl = baseUrl.TrimEnd('/');
            _apiPrefix = apiPrefix;
            _http = new HttpClient { Timeout = TimeSpan.FromSeconds(30) };
            _http.DefaultRequestHeaders.Add("Accept", "application/json");
            if (!string.IsNullOrEmpty(token))
                _http.DefaultRequestHeaders.Add("Authorization", $"Bearer {token}");
        }

        public async Task<string> GetAsync(string path, Dictionary<string, string>? parameters = null)
        {
            var url = _baseUrl + _apiPrefix + path;
            if (parameters is { Count: > 0 })
            {
                var query = HttpUtility.ParseQueryString(string.Empty);
                foreach (var kv in parameters) query[kv.Key] = kv.Value;
                url += "?" + query;
            }
            return await SendAsync(HttpMethod.Get, url, null);
        }

        public async Task<string> PostAsync(string path, string jsonBody)
        {
            var url = _baseUrl + _apiPrefix + path;
            return await SendAsync(HttpMethod.Post, url, jsonBody);
        }

        private async Task<string> SendAsync(HttpMethod method, string url, string? jsonBody)
        {
            using var request = new HttpRequestMessage(method, url);
            if (jsonBody != null)
                request.Content = new StringContent(jsonBody, Encoding.UTF8, "application/json");

            using var response = await _http.SendAsync(request);
            var body = await response.Content.ReadAsStringAsync();
            if (!response.IsSuccessStatusCode)
                throw new HttpRequestException(
                    $"ADIP HTTP {(int)response.StatusCode} for {method} {url}: {body}");
            return body;
        }

        // --- convenience helpers ---
        public Task<string> HealthAsync() => GetAsync("/health");

        public Task<string> ListProjectsAsync() =>
            GetAsync("/projects", new() { ["size"] = "50" });

        public Task<string> GenerateArtifactAsync(int projectId, string artifactType) =>
            GetAsync($"/artifact-generation/projects/{projectId}/generate",
                     new() { ["artifact_type"] = artifactType });

        public Task<string> OrchestrateAsync(string prompt) =>
            PostAsync("/orchestrator/run",
                      $"{{\"prompt\":\"{prompt.Replace("\"", "\\\"")}\"}}");

        public void Dispose() => _http.Dispose();
    }
}
